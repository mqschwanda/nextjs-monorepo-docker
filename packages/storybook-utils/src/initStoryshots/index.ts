import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
