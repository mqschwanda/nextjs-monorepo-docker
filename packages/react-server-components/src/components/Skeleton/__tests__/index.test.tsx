import path from 'path';
import initStoryshots from '@storybook/addon-storyshots';

initStoryshots({
  configPath: path.resolve(__dirname, '../../../../../storybook/.storybook'),
  storyKindRegex: /^@mqs\/react-server-components\/components\/Skeleton$/,
});
