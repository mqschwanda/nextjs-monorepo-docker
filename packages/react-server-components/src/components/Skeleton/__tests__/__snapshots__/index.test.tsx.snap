// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots @mqs/react-server-components/components/Skeleton Skeleton Card Example 1`] = `
<div
  className="card shadow-xl card-side card-normal"
  data-testid="Card"
>
  <figure>
    <div
      className="animate-pulse bg-neutral-content h-[200px] w-[200px]"
      data-testid="SkeletonDiv"
    />
  </figure>
  <div
    className="card-body"
    data-testid="CardBody"
  >
    <span
      className="card-title"
      data-testid="CardTitle"
    >
      <span
        className="animate-pulse bg-neutral-content inline-block rounded-md h-[1rem] w-[150px]"
        data-testid="SkeletonSpan"
      />
    </span>
    <p>
      <span
        className="animate-pulse bg-neutral-content inline-block rounded-md h-[1rem] w-full"
        data-testid="SkeletonSpan"
      />
    </p>
    <p>
      <span
        className="animate-pulse bg-neutral-content inline-block rounded-md h-[1rem] w-full"
        data-testid="SkeletonSpan"
      />
    </p>
    <p>
      <span
        className="animate-pulse bg-neutral-content inline-block rounded-md h-[1rem] w-full"
        data-testid="SkeletonSpan"
      />
    </p>
  </div>
</div>
`;
